packages/graphql/tests/integration/filtering/relationship-to-abstract-types.int.test.ts

```diff
@@ -0,0 +1,1171 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { faker } from "@faker-js/faker";
+import { graphql } from "graphql";
+import type { Driver, Session } from "neo4j-driver";
+import { generate } from "randomstring";
+import { Neo4jGraphQL } from "../../../src/classes";
+import { UniqueType } from "../../utils/graphql-types";
+import Neo4jHelper from "../neo4j";
+
+describe("interface relationships", () => {
+    let driver: Driver;
+    let neo4j: Neo4jHelper;
+    let session: Session;
+    let neoSchema: Neo4jGraphQL;
+
+    let typeMovie: UniqueType;
+    let typeSeries: UniqueType;
+    let typeActor: UniqueType;
+
+    beforeAll(async () => {
+        neo4j = new Neo4jHelper();
+        driver = await neo4j.getDriver();
+    });
+
+    beforeEach(async () => {
+        typeMovie = new UniqueType("Movie");
+        typeSeries = new UniqueType("Series");
+        typeActor = new UniqueType("Actor");
+        session = await neo4j.getSession();
+
+        const typeDefs = /* GraphQL */ `
+            interface Production {
+                title: String!
+            }
+
+            type ${typeMovie} implements Production {
+                title: String!
+                runtime: Int!
+            }
+
+            type ${typeSeries} implements Production {
+                title: String!
+                episodes: Int!
+            }
+
+            type ActedIn @relationshipProperties {
+                screenTime: Int!
+            }
+
+            type ${typeActor} {
+                name: String!
+                currentlyActingIn: Production @relationship(type: "CURRENTLY_ACTING_IN", direction: OUT)
+                actedIn: [Production!]! @relationship(type: "ACTED_IN", direction: OUT, properties: "ActedIn")
+            }
+        `;
+
+        neoSchema = new Neo4jGraphQL({
+            typeDefs,
+        });
+    });
+
+    afterEach(async () => {
+        await session.run(
```

Comment:
It would helpful to start using `cleanNodes` rather than manual deletion so that it's less error-prone and easier and consistent across all the teardown